### PR TITLE
chore(orca-api): move interfaces into api package

### PR DESCRIPTION
**Pull request by aelath**
This moves two interfaces from `orca-core` into `orca-api`.
- `ExecutionPreprocessor`
- `TaskExecutionInterceptor`

### CODE REVIEW

File: orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/ExecutionPreprocessor.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2019 Netflix, Inc.
+ * Copyright 2021 Armory
```
Comment (danielpeach) [inline orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/ExecutionPreprocessor.java]: I'm not exactly sure what the rules are here but my first thought was that this should not change (2019 -> 2021 seems OK though). 🤷

File: orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/TaskExecutionInterceptor.java [hunk 2]
```diff
@@ -14,11 +14,10 @@
  * limitations under the License.
  */
 
-package com.netflix.spinnaker.orca;
+package com.netflix.spinnaker.orca.api.pipeline;
 
-import com.netflix.spinnaker.orca.api.pipeline.Task;
-import com.netflix.spinnaker.orca.api.pipeline.TaskResult;
 import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution;
+
```
Comment (danielpeach) [inline orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/TaskExecutionInterceptor.java]: extraneous line

**Review by danielpeach — APPROVED** [review]
awesome

File: orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/ExecutionPreprocessor.java (hunk 1 above)
Comment (aelath) [reply orca-api/src/main/java/com/netflix/spinnaker/orca/api/pipeline/ExecutionPreprocessor.java]: Cool happy to leave this. 👍

**Review by claymccoy — APPROVED** [review]

**Review by claymccoy — APPROVED** [review]

**Comment by aelath** [comment]
Closing this as I made a cleaner version for OSS based on feedback: https://github.com/spinnaker/orca/pull/4121

**Pull request closed without merging**